### PR TITLE
[Feature][Transform] Add SimpleSQL transform plugin

### LINKED ISSUE #4149: [Feature][Transform-v2] Add a Simple SQL transform plugin with internal mem SQL engine
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

How about add a Simple SQL transform plugin with  internal mem SQL engine like H2DB, SQLite or Calcite.
We can via the transform plugin implements some SQL functions and criteria filter.

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by rewerma**
## Purpose of this pull request
We can use  simple SQL to transform input data. For example config:
```
transform {
  Sql {
    source_table_name = "fake"
    result_table_name = "fake1"
    query = "select id, concat(name, '_') as name, age+1 as age from fake where id>0"
  }
}
```

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [x] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [x] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by davidzollo** [comment]
The CI failed, please check and add license 
image

File: seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sqlengine/zeta/ZetaSimpleSQLEngine.java [hunk 1]
```diff
@@ -0,0 +1,568 @@
...
+        }
+        if (whereExpr instanceof Parenthesis) {
+            Parenthesis parenthesis = (Parenthesis) whereExpr;
+            Expression expression = parenthesis.getExpression();
+            return physicalFilter(expression, inputFields);
+        }
+        return false;
+    }
+
+    private Object[] physicalScanTable(SeaTunnelRow inputRow) {
```
Comment (Hisoka-X) [inline seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sqlengine/zeta/ZetaSimpleSQLEngine.java]: Seem like this logic shouldn't run on each row.

Comment (rewerma) [reply seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sqlengine/zeta/ZetaSimpleSQLEngine.java]: Yes，you are right

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/task/flow/TransformFlowLifeCycle.java [hunk 2]
```diff
@@ -93,4 +102,12 @@ public void received(Record<?> record) {
     public void restoreState(List<ActionSubtaskState> actionStateList) throws Exception {
         // nothing
     }
+
+    @Override
+    public void close() throws IOException {
+        for (SeaTunnelTransform<T> t : transform) {
+            t.close();
```
Comment (Hisoka-X) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/task/flow/TransformFlowLifeCycle.java]: Hi, please catch close method on each transfrom, so we can make sure every transform close method can be called.

Comment (rewerma) [reply seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/task/flow/TransformFlowLifeCycle.java]: ok

**Comment by TaoZex** [comment]
Good Job!
Please add this feature in release-note.

**Comment by rewerma** [comment]
> Good Job! Please add this feature in release-note.

ok

**Comment by Hisoka-X** [comment]
Hi, maybe change name `SimpleSQL` to `SQL` will be better. Clear, easy to understand.

**Comment by rewerma** [comment]
> Hi, maybe change name `SimpleSQL` to `SQL` will be better. Clear, easy to understand.

The plugin name `SQL` is the same name as the old transform of Spark and Flink engine? Is the old SQL transform plugin removed？

**Comment by Hisoka-X** [comment]
> > Hi, maybe change name `SimpleSQL` to `SQL` will be better. Clear, easy to understand.
> 
> The plugin name `SQL` is the same name as the old transform of Spark and Flink engine? Is the old SQL transform plugin removed？

yes

**Review by hailin0 — APPROVED** [review]

**Review by ashulin — APPROVED** [review]
LGTM

**Pull request merged**